specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -3562,6 +3562,462 @@
+          }
+        },
+        "tags": [
+          "ThreatIntelligence"
+        ],
+        "description": "Patch a threat intelligence.",
+        "operationId": "ThreatIntelligenceIndicator_Update",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/OperationalInsightsResourceProvider"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/ThreatIntelligenceIdentifier"
+          },
+          {
+            "$ref": "#/parameters/ThreatIntelligenceIndicatorObjectToPatch"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ThreatIntelligenceResource"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ThreatIntelligenceResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete the Threat Intelligence": {
+            "$ref": "./examples/threatintelligence/DeleteThreatIntelligence.json"
+          }
+        },
+        "tags": [
+          "ThreatIntelligence"
+        ],
+        "description": "Delete a threat intelligence.",
+        "operationId": "ThreatIntelligenceIndicator_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/OperationalInsightsResourceProvider"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/ThreatIntelligenceIdentifier"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/threatIntelligence/query": {
```

Comment:
the current API structure will not work with ARM. the segments are incorrect. POST action is over an instance of a type.
If you want to provide query capability, may be expose it as a provider level action POST .../microsoft.securityINsights/queryThreatINtelligence. 
RBAC can be violated via this API though, someone who does not have /read permission on threatIntelligence and just has permission to call this POST action can read all the resources.

Reply:
Hey, we are working with Rbac team and adding query/action to the Reader role as well, so this won't be an issue

Reply:
but when ARM parses the segment for this particular API, it will not be able to map it to what you are planning on adding by working with RBAC team.  ARM expects segments like {resourceType}/{resourceTypeInstance}/{actionName}
And the authorization action for above will become {resourceType}/{actionName}/action
You are missing the "resourceTypeInstance" segment.

Reply:
Gaurav's comment remains unaddressed here.

---
In reply to: [474471959](https://github.com/Azure/azure-rest-api-specs/pull/10463#discussion_r474471959) [](ancestors = 474471959)